src/lib.rs

```diff
@@ -105,11 +126,188 @@ impl Fairing for RocketSentry {
                 if config.sentry_dsn.is_empty() {
                     info!("Sentry disabled.");
                 } else {
-                    self.init(&config.sentry_dsn);
+                    let traces_sample_rate = config.sentry_traces_sample_rate.unwrap_or(0f32);
+                    self.init(&config.sentry_dsn, traces_sample_rate);
                 }
             }
             Err(err) => error!("Sentry not configured: {}", err),
         }
         Ok(rocket)
     }
+
+    async fn on_request(&self, request: &mut Request<'_>, _: &mut Data<'_>) {
```

Comment:
Since this introduces some overhead on request/response, let's short circuit and skip all this logic if Sentry is disabled or if sample rate is `0`.

Maybe add `transactions_enabled: AtomicBool` member to `struct RocketSentry` that gets set in init/ignition.

Reply:
Great idea!